### PR TITLE
mautrix-signal: init at 0.1.1

### PR DESCRIPTION
**Pull request by expipiplus1**
The script wrapping python -m mautrix_signal was inspired by
the gunicorn script in powerdns-admin

@ma27 or @nyanloutre (maintaners of the mautrix-telegram package): would you like to be maintainers too?

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
  - (doesn't crash immediately)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by expipiplus1** [comment]
I've tested this up to the point of writing a config file. I suspect it won't operate usefully until someone packages signald https://gitlab.com/signald/signald/

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119285` at ba9fa9a3 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>mautrix-signal</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/servers/mautrix-signal/default.nix:54:0:
```
   |
54 |     description = "A Matrix-Signal puppeting bridge";
   | ^
```
</li>
  </ul>
</details>

**Comment by Ma27** [comment]
First of all, can you please remove at least my name from the commit before merging this? The problem is that as soon as somebody rebases master or cherry-picks the commit to somewhere else or whatever, I get a notification.

While I'm happy to help if there are problems, I guess I have to decline the offer since I'm not using signal (or the bridge) and don't think that this will change soon, but thanks for asking! :)

**Comment by expipiplus1** [comment]
> First of all, can you please remove at least my name from the commit before merging this? The problem is that as soon as somebody rebases master or cherry-picks the commit to somewhere else or whatever, I get a notification.

Oops! done.

> While I'm happy to help if there are problems, I guess I have to decline the offer since I'm not using signal (or the bridge) and don't think that this will change soon, but thanks for asking! :)

:+1:

File: pkgs/servers/mautrix-signal/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{ lib, python3, fetchFromGitHub }:
+
+with python3.pkgs;
+
+buildPythonPackage rec {
+  pname = "mautrix-signal";
+  version = "v0.1.1";
```
Comment (SuperSandro2000) [inline pkgs/servers/mautrix-signal/default.nix:7]:
```suggestion
  version = "0.1.1";
```

File: pkgs/servers/mautrix-signal/default.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  src = fetchFromGitHub {
+    owner = "tulir";
+    repo = "mautrix-signal";
+    rev = "39f5d39e8cb16538cf91304c2782eec0d0df38a6";
```
Comment (SuperSandro2000) [inline pkgs/servers/mautrix-signal/default.nix:12]:
```suggestion
    rev = "v${version}";
```

File: pkgs/servers/mautrix-signal/default.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  mautrixSignalScript = ''
+    #!/bin/sh
+    exec python -m mautrix_signal "$@"
+  '';
+
+  postInstall = ''
+    mkdir -p $out/bin
+
+    # Make a little wrapper for running mautrix-signal with its dependencies
+    echo "$mautrixSignalScript" > $out/bin/mautrix-signal
```
Comment (SuperSandro2000) [inline pkgs/servers/mautrix-signal/default.nix:45]:
```suggestion
    echo "#!/bin/sh
      exec python -m mautrix_signal \"$@\"
    " > $out/bin/mautrix-signal
```

File: pkgs/servers/mautrix-signal/default.nix [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    unpaddedbase64
+    yarl
+  ];
+
+  doCheck = false;
+
+  mautrixSignalScript = ''
+    #!/bin/sh
+    exec python -m mautrix_signal "$@"
+  '';
```
Comment (SuperSandro2000) [inline pkgs/servers/mautrix-signal/default.nix:39]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by expipiplus1** [comment]
Thanks for the review, I've applied the changes.

I've also added signald which is necessary to use mautrix-signal.

I have a couple of modules set up to use these packages but I think I'll leave them to another PR

File: pkgs/servers/mautrix-signal/default.nix [hunk 5]
```diff
@@ -0,0 +1,57 @@
+{ lib, python3, fetchFromGitHub }:
+
+with python3.pkgs;
+
```
Comment (SuperSandro2000) [inline pkgs/servers/mautrix-signal/default.nix:4]:
```suggestion
```

File: pkgs/servers/mautrix-signal/default.nix [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "tulir";
+    repo = "mautrix-signal";
+    rev = "v${version}";
+    sha256 = "11snsl7i407855h39g1fgk26hinnq0inr8sjrgd319li0d3jwzxl";
+  };
+
+  propagatedBuildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/servers/mautrix-signal/default.nix:16]:
```suggestion
  propagatedBuildInputs = with python3.pkgs; [
```

File: pkgs/applications/networking/instant-messengers/signald/default.nix [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+      unofficial, unapproved, and not nearly as secure as the real Signal
+      clients.
+    '';
+    homepage = "https://signald.org";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ expipiplus1 ];
+    platforms = platforms.unix;
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/signald/default.nix:87]: One final new line to much.

File: pkgs/applications/networking/instant-messengers/signald/default.nix [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    sed -i -e 's|mavenCentral()|mavenLocal(); maven { url uri("${deps}") }|' build.gradle
+
+    gradle --offline --no-daemon distTar
+  '';
+
+  installPhase = ''
+    mkdir -p $out
+    tar xvf ./build/distributions/signald.tar --strip-components=1 --directory $out/
+    wrapProgram $out/bin/signald \
+      --prefix PATH : "${coreutils}/bin" \
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/signald/default.nix:66]:
```suggestion
      --prefix PATH : ${makeBinPath [ coreutils ]} \
```

File: pkgs/applications/networking/instant-messengers/signald/default.nix [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+  src = fetchgit {
+    url = "https://gitlab.com/signald/signald";
+    rev = version;
+    sha256 = "1ilmg0i1kw2yc7m3hxw1bqdpl3i9wwbj8623qmz9cxhhavbcd5i7";
+    leaveDotGit = true;
+  };
+
+  buildConfigJar = fetchurl {
+    url =
+      "https://dl.bintray.com/mfuerstenau/maven/gradle/plugin/de/fuerstenau/BuildConfigPlugin/1.1.8/BuildConfigPlugin-1.1.8.jar";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/signald/default.nix:17]:
```suggestion
    url = "https://dl.bintray.com/mfuerstenau/maven/gradle/plugin/de/fuerstenau/BuildConfigPlugin/1.1.8/BuildConfigPlugin-1.1.8.jar";
```

Isn't bintray moving downloads?

File: pkgs/applications/networking/instant-messengers/signald/default.nix [hunk 10]
```diff
@@ -0,0 +1,87 @@
+{ lib, stdenv, fetchurl, fetchgit, jre, coreutils, gradle_6, git, perl
+, makeWrapper }:
+
+let
+  pname = "signald";
+
+  version = "0.13.1";
+  src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/signald/default.nix:8]:
```suggestion
  version = "0.13.1";

  src = fetchgit {
```

Also please use fetchFromGitLab.

File: pkgs/applications/networking/instant-messengers/signald/default.nix (hunk 9 above)
Comment (expipiplus1) [reply pkgs/applications/networking/instant-messengers/signald/default.nix:17]: I think there's 1y grace maybe?

I couldn't find an alternate location, but have raised an upstream issue.

File: pkgs/applications/networking/instant-messengers/signald/default.nix (hunk 10 above)
Comment (expipiplus1) [reply pkgs/applications/networking/instant-messengers/signald/default.nix:8]: This package uses the .git directory sadly. I'll make a comment

**Comment by expipiplus1** [comment]
Possible to merge this, not sure what to do about the bintray thing.

File: pkgs/applications/networking/instant-messengers/signald/default.nix [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+    sed -i -e 's|mavenCentral()|mavenLocal(); maven { url uri("${deps}") }|' build.gradle
+
+    gradle --offline --no-daemon distTar
+  '';
+
+  installPhase = ''
+    mkdir -p $out
+    tar xvf ./build/distributions/signald.tar --strip-components=1 --directory $out/
+    wrapProgram $out/bin/signald \
+      --prefix PATH : ${makeBinPath [ coreutils ]} \
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/signald/default.nix:67]:
```suggestion
      --prefix PATH : ${lib.makeBinPath [ coreutils ]} \
```

Comment (expipiplus1) [reply pkgs/applications/networking/instant-messengers/signald/default.nix:67]: done

File: pkgs/applications/networking/instant-messengers/signald/default.nix [hunk 12]
```diff
@@ -0,0 +1,87 @@
+{ lib, stdenv, fetchurl, fetchgit, jre, coreutils, gradle_6, git, perl
+, makeWrapper }:
+
+let
+  pname = "signald";
+
+  version = "0.13.1";
+
+  # This package uses the .git directory
+  src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/signald/default.nix:10]: Cna this use fetchFromGitLab?

**Comment by expipiplus1** [comment]
The build requires the .git directory sadly

**Comment by SuperSandro2000** [comment]
> The build requires the .git directory sadly

Can we patch this? fetchFromGitLab should also support leaveDotGit.

**Comment by expipiplus1** [comment]
> fetchFromGitLab should also support leaveDotGit

I don't think it does sadly.

It's not impossible to patch, but it's just exchanging cheap computer time for my valuable human time.

**Comment by SuperSandro2000** [comment]
> it's just exchanging cheap computer time for my valuable human time.

We still don't want to be more wasteful than we need to be.

**Pull request merged**